### DISCUSSION TITLE
`x-mask`: add support for zero padding

### DISCUSSION THREAD

**Discussion by ajschmidt8** [alpinejs/alpine#3633] [3. Feature ideas]
It'd be nice if the `x-mask` plugin supported zero padding for money.

For instance, if a user types in `25`, `x-mask` could pad it to `25.00`.

I'm happy to work on this if someone could provide some guidance.

It wasn't immediately clear to me how to implement this based on a cursory glance at the existing `x-mask` code.

**Comment by ekwoka** [comment]
should this allow `24.99` or not?

I think clarifying what the actual behavior you want is would help in understanding how to support it and even if x-mask is the right thing.

If you just want to let the user type whatever and update it to be decimal when they are done, just a blur event handler would be enough.

```html
x-model="price" @blur="price = Number(price).toFixed(2)"
```

**Reply by ajschmidt8** [reply]
That was exactly what I was looking for.

Thank you!

**Reply by eznix86** [reply]
How to make work with the $money() ?

**Reply by ekwoka** [reply]
Doesn't money already do this?
